dist/IO/t/io_unix.t

```diff
@@ -1,28 +1,43 @@
 #!./perl
 
 use Config;
+use IO::Socket;
 
 BEGIN {
     my $reason;
+    my $can_fork = $Config{d_fork} ||
+		    (($^O eq 'MSWin32' || $^O eq 'NetWare') and
+		     $Config{useithreads} and
+		     $Config{ccflags} =~ /-DPERL_IMPLICIT_SYS/
+		    );
+
     if ($ENV{PERL_CORE} and $Config{'extensions'} !~ /\bSocket\b/) {
 	$reason = 'Socket extension unavailable';
     }
     elsif ($ENV{PERL_CORE} and $Config{'extensions'} !~ /\bIO\b/) {
 	$reason = 'IO extension unavailable';
     }
     elsif ($^O eq 'os2') {
-	require IO::Socket;
-
 	eval {IO::Socket::pack_sockaddr_un('/foo/bar') || 1}
 	  or $@ !~ /not implemented/ or
 	    $reason = 'compiled without TCP/IP stack v4';
     }
-    elsif ($^O =~ m/^(?:qnx|nto|vos|MSWin32)$/ ) {
+    elsif ($^O =~ m/^(?:qnx|nto|vos)$/ ) {
 	$reason = "UNIX domain sockets not implemented on $^O";
     }
-    elsif (! $Config{'d_fork'}) {
+    elsif (! $can_fork) {
 	$reason = 'no fork';
     }
+    elsif ($^O eq 'MSWin32') {
+      if ($ENV{CONTINUOUS_INTEGRATION}) {
+         $reason = 'Skipping on Windows CI, see gh17575 and gh17429';
+      } else {
+       eval { socket(my $sock, PF_UNIX, SOCK_STREAM, 0) or die "AF_UNIX unsupported" }
+         or $@ !~ /AF_UNIX unsupported/
+         or $reason = "AF_UNIX support either unavailable or disabled on this platform";
```

Comment:
this seems hard to read... could not this be simplified using something like this
`$reason = "..." unless eval { socket(my $sock, PF_UNIX, SOCK_STREAM, 0); 1 }    `

Reply:
Yeah, I copied an existing, nearby eval check for the sake of visual consistency, but am happy to revise.

Reply:
thanks